tests/components/generic/test_camera.py

```diff
@@ -156,6 +158,104 @@ async def test_limit_refetch(aioclient_mock, hass, hass_client):
     assert body == "hello planet"
 
 
+async def test_stream_source(aioclient_mock, hass, hass_client, hass_ws_client):
+    """Test that the stream source is rendered."""
+    assert await async_setup_component(
+        hass,
+        "camera",
+        {
+            "camera": {
+                "name": "config_test",
+                "platform": "generic",
+                "still_image_url": "https://example.com",
+                "stream_source": 'http://example.com/{{ states.sensor.temp.state + "a" }}',
+                "limit_refetch_to_url_change": True,
+            }
+        },
+    )
+    await hass.async_block_till_done()
+
+    hass.states.async_set("sensor.temp", "5")
+
+    with patch(
+        "homeassistant.components.camera.request_stream",
+        return_value="http://home.assistant/playlist.m3u8",
+    ) as mock_request_stream:
+        # Request playlist through WebSocket
+        client = await hass_ws_client(hass)
+
+        await client.send_json(
+            {"id": 1, "type": "camera/stream", "entity_id": "camera.config_test"}
+        )
+        msg = await client.receive_json()
+
+        # Assert WebSocket response
+        assert mock_request_stream.call_count == 1
+        assert mock_request_stream.call_args[0][1] == "http://example.com/5a"
+        assert msg["id"] == 1
+        assert msg["type"] == TYPE_RESULT
+        assert msg["success"]
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5737505)